engine/crates/engine-v2/src/engine.rs

```diff
@@ -435,28 +455,46 @@ impl<'ctx, R: Runtime> PreExecutionContext<'ctx, R> {
         };
 
         let mut status: GraphqlResponseStatus = GraphqlResponseStatus::Success;
+
         struct Sender<'a> {
             sender: mpsc::Sender<Response>,
             status: &'a mut GraphqlResponseStatus,
+            operation_name: Option<String>,
+            client: Option<Client>,
+            operation_metrics: GraphqlOperationMetrics,
```

Comment:
why clone? could just be a `&GraphqlOperationMetrics`

Reply:
Yeah, we could, if the async block starting from line 430 would not take the ownership of self...

Reply:
`self` is retrieved back as `ctx`, so you can just do `ctx.engine.operation_metrics`